techbuzz/src/main/java/com/sivalabs/techbuzz/posts/web/controllers/AddVoteController.java

```diff
@@ -28,6 +38,48 @@ class AddVoteController {
     public ResponseEntity<Void> createVote(@Valid @RequestBody CreateVoteRequest request, @CurrentUser User loginUser) {
         var createVoteRequest = new CreateVoteRequest(request.postId(), loginUser.getId(), request.value());
         postService.addVote(createVoteRequest);
+
         return ResponseEntity.ok().build();
     }
+
+    @GetMapping(path = "/api/hx/votes/update/{voteType}/{postId}", produces = "text/html")
+    @ResponseBody
+    public String createVoteHTMX(
+            @PathVariable String postId, @PathVariable String voteType, @CurrentUser User loginUser) {
+        CreateVoteRequest createVoteRequest =
+                new CreateVoteRequest(Long.valueOf(postId), loginUser.getId(), Integer.valueOf(voteType));
+        postService.addVote(createVoteRequest);
+        Post updatedPost = postService.getPost(Long.valueOf(postId));
+        PostViewDTO postViewDTO = postMapper.toPostViewDTO(loginUser, updatedPost);
+        long downVoteCount = postViewDTO.getDownVoteCount();
+        long upVoteCount = postViewDTO.getUpVoteCount();
+
+        StringBuilder returnHtml = new StringBuilder();
+        returnHtml.append("<div id=\"vote-container-" + postId + "\">");
```

Comment:
We can have a thymeleaf template instead of creating the HTML snippet using StringBuilder.